packages/mui-joy/src/ChipDelete/ChipDelete.tsx

```diff
@@ -101,6 +107,29 @@ const ChipDelete = React.forwardRef(function ChipDelete(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
+  const handleOnDelete = (
+    event: React.KeyboardEvent<HTMLButtonElement> | React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    if ('key' in event) {
```

Comment:
Does this check happen only when a key is pressed?

Reply:
yes , "key" property is not present in mouse event

Reply:
Would be better to separate the `handleDelete` into 2 functions for `onClick` and `onKeyDown`:

```js
const handleClickDelete = () => ...

const handleKeyDelete = () => ...
```

Reply:
So you don't have to check the `'key' in event`. It looks cleaner to me.